grails-app/services/com/mini/asaas/CustomerService.groovy

```diff
@@ -0,0 +1,168 @@
+package com.mini.asaas
+
+import core.dtos.AddressDTO
+import core.dtos.CustomerDTO
+import core.exceptions.EntityNoDataChangedException
+import core.exceptions.EntityNotFoundException
+import core.exceptions.EntityWithSamePropertyAlreadyExistsException
+import grails.compiler.GrailsCompileStatic
+import grails.gorm.transactions.Transactional
+import grails.validation.ValidationException
+
+/**
+ * Serviço responsável por gerenciar as operações de {@code Customer}.
+ */
+@GrailsCompileStatic
+@Transactional
+class CustomerService {
+
+    /**
+     * Cria um novo {@code Customer} a partir dos dados pessoais e de endereço informados.
+     *
+     * @param customerDTO Dados pessoais do {@code Customer}.
+     * @param addressDTO Dados de endereço do {@code Customer}.
+     * @return {@code Customer} criado.
+     *
+     * @throws EntityWithSamePropertyAlreadyExistsException Se já existir um {@code Customer} com o mesmo e-mail ou CPF/CNPJ.
+     * @throws ValidationException Se ocorrer um erro ao validar os dados pessoais.
+     *
+     * @see CustomerDTO
+     * @see AddressDTO
+     * @see Customer
+     * @see Customer#address
+     */
+    public Customer save(CustomerDTO customerDTO, AddressDTO addressDTO) {
+
+        if (Customer.findByCpfCnpj(customerDTO.cpfCnpj)) {
+            throw new EntityWithSamePropertyAlreadyExistsException("Já existe um cliente cadastrado com o CPF/CNPJ informado")
+        }
+
+        if (Customer.findByEmail(customerDTO.email)) {
+            throw new EntityWithSamePropertyAlreadyExistsException("Já existe um cliente cadastrado com o e-mail informado")
+        }
+
+        Customer customer = customerDTO.toCustomer()
+        customer.address = addressDTO.toAddress()
+
+        if (!customer.validate()) {
```

Comment:
isso aqui não costuma ser uma abordagem comum, isso vai validar os dados conforme oque foi definido pra entidade, porém, caberiam bem mais validações quando falamos de regra de negócio, no Asaas não usamos isso por exemplo... O padrão é ter um método validate que valida preenchimento, formatação correta, e tudo mais que for necessário pra manter a consistência desse cadastro. Se mesmo assim algo passar na validação, terá erro ao salvar e é importante a controller estar preparada pra lidar e retornar algo a respeito